libs/@blockprotocol/core/src/core-embedder-handler.ts

```diff
@@ -61,11 +63,18 @@ export class CoreEmbedderHandler extends CoreHandler {
     this.updateDispatchElementFromEvent(event);
 
     // get the properties sent on initialization for any registered services
-    const data: EmbedderInitMessage = {};
-    for (const [serviceName, serviceInstance] of this.services) {
-      data[serviceName] = serviceInstance.getInitPayload();
+    let data = this.initResponse;
+
+    if (!data) {
+      data = {};
+
+      for (const [serviceName, serviceInstance] of this.services) {
+        data[serviceName] = serviceInstance.getInitPayload();
+      }
     }
```

Comment:
What was the motivation behind this caching? I'm wondering if it's worth it – in theory, the payload could become out of alignment with the properties it is derived from. It might be better to just let the service provide the freshest version of its init payload when needed. 

Is it correct to say that this would only ever be called twice by a block that isn't using the new `CoreBlockHandler` implementation? Since it will only ever send `init` once

Reply:
Safety/worries about spec compliance, mostly. I wasn't sure of the possible consequences of a service being initiated multiple times, so I wanted safety on both ends this wouldn't happen. Additionally because of the timing uncertainties around sandboxing, it might actually be possible the init message gets sent by `CoreBlockHandler` multiple times (as it gets sent a second time before the response to the first message gets back) so I wanted to create more certainty around how it would behave.